### ISSUE TITLE
Add a jsonschema for charmcraft.yaml in jsonstore

### ISSUE THREAD

**Issue by facundobatista** [canonical/charmcraft#687]
The idea is that editors could retrieve automatically the schema from [the json schema store](https://www.schemastore.org/json/) and validate that charmcraft.yaml makes sense.

**Comment by sed-i** [comment]
@lengau, with the unified charmcraft.yaml this would make a lot of sense!
This would also obviate https://github.com/juju/charm/pull/370.

**Comment by lengau** [comment]
I agree! We're migrating the full `charmcraft.yaml` description to pydantic models, after which we should be able to [automatically generate the JSON schema](https://docs.pydantic.dev/latest/concepts/json_schema/). This will need some additional work in craft-parts as well, but that work would also be reusable

**Comment by lengau** [comment]
Note to the implementer, this would include both creating the schema and PRing a relevant change to the schema store. See: https://github.com/SchemaStore/schemastore/pull/668
